src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -939,49 +969,32 @@ def post_process(self, r):
+            # Case: Kafka on Confluent Cloud. Update list with 2-parameter connectors
+            service_connectors_def_list = update_connectors_with_two_parameters(service_connectors_def_list)
+
+            check_bindings_and_raise_error(self.cmd, service_connectors_def_list, service_bindings_def_list, self.get_argument_resource_group_name(), self.get_argument_name())
+            # Only set these for post_process if no errors were raised
             self.set_argument_service_connectors_def_list(service_connectors_def_list)
-            service_bindings_used_map = {update_item["name"]: False for update_item in service_bindings_def_list}
+            self.set_argument_service_bindings_def_list(service_bindings_def_list)
 
             safe_set(self.new_containerapp, "properties", "template", "serviceBinds", value=self.containerapp_def["properties"]["template"]["serviceBinds"])
 
             if self.new_containerapp["properties"]["template"]["serviceBinds"] is None:
                 self.new_containerapp["properties"]["template"]["serviceBinds"] = []
 
-            for item in self.new_containerapp["properties"]["template"]["serviceBinds"]:
-                for update_item in service_bindings_def_list:
-                    if update_item["name"] in item.values():
-                        item["serviceId"] = update_item["serviceId"]
-                        service_bindings_used_map[update_item["name"]] = True
-
             for update_item in service_bindings_def_list:
-                if service_bindings_used_map[update_item["name"]] is False:
-                    # Check if it doesn't exist in existing service linkers
-                    managed_bindings = linker_client.linker.list(resource_uri=self.containerapp_def["id"])
-                    if managed_bindings:
-                        managed_bindings_list = [item.name for item in managed_bindings]
-                        if update_item["name"] in managed_bindings_list:
-                            raise ValidationError("Binding names across managed and dev services should be unique.")
-
-                    self.new_containerapp["properties"]["template"]["serviceBinds"].append(update_item)
-
-            if service_connectors_def_list is not None:
-                for item in service_connectors_def_list:
-                    # Check if it doesn't exist in existing service bindings
-                    service_bindings_list = []
-                    for binds in self.new_containerapp["properties"]["template"]["serviceBinds"]:
-                        service_bindings_list.append(binds["name"])
-                        if item["linker_name"] in service_bindings_list:
-                            raise ValidationError("Binding names across managed and dev services should be unique.")
+                self.new_containerapp["properties"]["template"]["serviceBinds"].append(update_item)
```

Comment:
May I ask why to delete this logic here?
In `az containerapp update --bind`, if the containerapp already bind service `x`, then update `x`, seems append directly will cause duplicate service `x` in `serviceBinds `list.
`safe_set(self.new_containerapp, "properties", "template", "serviceBinds", value=self.containerapp_def["properties"]["template"]["serviceBinds"])`

Reply:
I realized that `check_bindings_and_raise_error` already takes care of that. I didn't add the initial code that i'm now removing but upon review, the `check_unique_bindings` functions that `check_bindings_and_raise_error` calls will ensure no duplicate bindings exist.

Reply:
What will happen if run `az containerapp update --bind redistest` twice?

Reply:
with the way in my old commit, it would throw an error bc the app is already bound. To keep up with how update works for other commands, it makes sense that it runs through, though. i have restored the logic to reflect this!